sdk/communication/communication-sms/src/optOutsClient.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { tracingClient } from "./generated/src/tracing.js";
+import type { OptOutResponseItem } from "./generated/src/index.js";
+import type { SmsApiClient } from "./generated/src/smsApiClient.js";
+import { generateOptOutRequest } from "./utils/smsUtils.js";
+import { extractOperationOptions } from "./extractOperationOptions.js";
+import type { OperationOptions } from "@azure/core-client";
+
+export interface OptOutCheckResult {
+  to: string;
+  isOptedOut: boolean;
+  httpStatusCode: number;
```

Comment:
I am not familiar with ACS so please excuse ny ignorance here but I wonder why the http status code is part of the response model? The user can access http info using the onResponse call back function in the operation options.

Reply:
These are batch requests where every item inside an array is processed separately and therefore may have a different status code